File: csharp/011_Container_With_Most_Water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int maxWater = 0;

        int left = 0, right = height.Length - 1;
        while (left < right) {
            int currWater = Math.Min(height[left], height[right]) * (right - left);
            maxWater = Math.Max(maxWater, currWater);
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return maxWater;
    }
}

File: cpp/011_Container_With_Most_Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxArea = 0;

        size_t left = 0;
        size_t right = height.size() - 1;

        int a, b;

        // area = WxH -> loop throught max W to 0 to find optimal H
        while (left < right) {
            int currArea = min(height[right], height[left]) * (right - left);
            maxArea = max(maxArea, currArea);
            if (height[left] < height[right])
                left++;
            else
                right--;
        }

        return maxArea;
    }
};